Skyra.Core/Cache/Models/Prompts/CoreRichDisplay.cs

```diff
@@ -0,0 +1,315 @@
+				case CoreRichDisplayReactionType.Last:
+					if (PagePosition == Context.Length - 1) return TimeSpan.Zero;
+					PagePosition = Context.Length - 1;
+					if (await Render(reaction.ChannelId)) return TimeSpan.FromMinutes(10);
+					return null;
+				case CoreRichDisplayReactionType.Info:
+					if (PagePosition == -1) return TimeSpan.Zero;
+					PagePosition = -1;
+					if (await Render(reaction.ChannelId)) return TimeSpan.FromMinutes(10);
+					return null;
+				case CoreRichDisplayReactionType.Stop:
+					// await RemoveReactions(reaction.ChannelId);
+					return null;
+				case CoreRichDisplayReactionType.Jump:
+					// TODO(kyranet): Create text prompt to ask for the number
+					// TODO(kyranet): Render
+					return TimeSpan.FromMinutes(10);
+				default:
+					throw new ArgumentOutOfRangeException();
+			}
+		}
+
+		[NotNull]
+		public CoreRichDisplay SetEmojis(CoreRichDisplayEmojis emojis)
+		{
+			Emojis = emojis;
+			return this;
+		}
+
+		[NotNull]
+		public CoreRichDisplay SetFooterPrefix(string prefix)
+		{
+			FooterEnabled = false;
+			FooterPrefix = prefix;
+			return this;
+		}
+
+		[NotNull]
+		public CoreRichDisplay SetFooterSuffix(string suffix)
+		{
+			FooterEnabled = false;
+			FooterSuffix = suffix;
+			return this;
+		}
+
+		[NotNull]
+		public CoreRichDisplay UseCustomFooters()
+		{
+			FooterEnabled = true;
+			return this;
+		}
+
+		[NotNull]
+		public CoreRichDisplay AddPage(CoreMessageEmbed embed)
+		{
+			Context = Context.Append(embed).ToArray();
+			return this;
+		}
+
+		[NotNull]
+		public CoreRichDisplay AddPage([NotNull] Func<CoreMessageEmbed, CoreMessageEmbed> callback)
+		{
+			var value = callback(Template);
+			Context = Context.Append(value).ToArray();
+			return this;
+		}
+
+		[NotNull]
+		public CoreRichDisplay SetInformationPage(CoreMessageEmbed? embed)
+		{
+			InformationPage = embed;
+			return this;
+		}
+
+		private async Task<bool> RemoveReactions(string channelId)
+		{
+			var reactionDeleteQuery =
+				IClient.Instance.Rest.Channels[channelId].Messages[MessageId.ToString()].Reactions.DeleteAsync();
+
+			return await Utilities.ResolveAsBooleanOnErrorCodes(reactionDeleteQuery, IgnoreMessageUpdateCodes);
+		}
+
+		private async Task<bool> Render(string channelId)
+		{
+			var messageUpdateQuery = IClient.Instance.Rest.Channels[channelId]
+				.Messages[MessageId.ToString()].PatchAsync(new SendableMessage
+				{
+					Embed = PagePosition == -1 ? InformationPage : Context[PagePosition]
+				});
+
+			return await Utilities.ResolveAsBooleanOnErrorCodes(messageUpdateQuery, IgnoreMessageUpdateCodes);
+		}
+
+		private CoreRichDisplayReactionType RetrieveEntry(string emoji)
+		{
+			foreach (var (type, allowedEmoji) in AllowedEmojis)
+			{
+				if (allowedEmoji == emoji) return type;
+			}
+
+			return CoreRichDisplayReactionType.None;
+		}
+
+		[ItemNotNull]
+		public async Task<CoreRichDisplay> SetUpAsync([NotNull] CoreMessage message,
+			CoreRichDisplayRunOptions? options = null)
```

Comment:
instead of using null for the default here, why not use `options = new CoreRichDisplayOptions();` to save the assignment and check below?

Reply:
Because it's not possible to do that as it's a class (structs weren't working for me).

Reply:
TIL.